core/translate/where_clause.rs

```diff
@@ -173,49 +187,71 @@ pub fn translate_processed_where<'a>(
             continue;
         }
         let target_jump = current_loop.next_row_label;
-        translate_condition_expr(program, select, &term.expr, target_jump, false, cursor_hint)?;
+        let success_label = program.allocate_label();
+        translate_condition_expr(
+            program,
+            select,
+            &term.expr,
+            cursor_hint,
+            ConditionMetadata {
+                jump_on_success: false,
+                on_failure_jump_label: target_jump,
+                on_success_jump_label: success_label,
+            },
+        )?;
+
+        program.resolve_label(success_label, program.offset());
     }
 
     Ok(())
 }
 
+#[derive(Default, Debug, Clone, Copy)]
+struct ConditionMetadata {
+    jump_on_success: bool,
+    on_success_jump_label: BranchOffset,
```

Comment:
Can we call these `jump_if_true_label` and `jump_if_false_label`? Success and failure suggest relation to error handling.

Reply:
Yeah I thought about this before falling asleep last night and woke up with the same thought, so I will do it :D